### PR TITLE
[Doc] Fix some minor markup errors in manual

### PR DESCRIPTION
**Pull request by quantimnot**
* Fix header level for noalias pragma section.
* Fix code snippet outside of code block that raised an error with `rst2html`.
* Fix broken 'Convertible relation' links that were raising warnings.

### CODE REVIEW

**Comment by a-mr** [comment]
It's worth noting that all automatic links to subsections (like `` `convertible relations`_ `` ) are broken in cases when there is a TOC in a file. But I would not bother fixing them at the moment (this PR is different though because there is that annoying warning). It's a regression since 1.6.x, probably broken by my addition of smart doc links #18642.

We shall to convert them to Markdown style links anyway. Markdown spec allows only links with the explicit target, but fortunately there is [the implicit_header_references extension](https://pandoc.org/MANUAL.html#extension-implicit_header_references) from Pandoc Markdown.

To summarize:

```markdown
Type relations
==============

Convertible relation
--------------------

Old RST ways of referencing:
- manual `Convertible relation <#type-relations-convertible-relation>`_
- automatic, buggy in devel  `Convertible relation`_ 

New Markdown ways:
- manual [Convertible relation](#type-relations-convertible-relation)
- automatic, not yet supported: [Convertible relation]()
```

**Comment by quantimnot** [comment]
@a-mr I thought there was some automatic mechanism and I tried removing the explicit anchor in the first link, but it wasn't working. Good to know about the changes. Thank you

**Review by Varriount — APPROVED** [review]

**Comment by a-mr** [comment]
Actually I re-checked links to sub-headers in 1.6 and 1.0 and it did not work.
So it's unrelated with recent changes and very probably never worked.
